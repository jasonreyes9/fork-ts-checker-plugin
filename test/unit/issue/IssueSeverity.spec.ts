import {
  IssueSeverity,
  isIssueSeverity,
  compareIssueSeverities
} from '../../../lib/issue';

describe('[UNIT] issue/IssueSeverity', () => {
  it('defines issue severity enum', () => {
    expect(IssueSeverity).toMatchSnapshot();
  });

  it.each([IssueSeverity.ERROR, IssueSeverity.WARNING])(
    "checks if '%p' is a IssueSeverity",
    severity => {
      expect(isIssueSeverity(severity)).toEqual(true);
    }
  );

  it.each([null, undefined, '', 'test', 1, {}, new Date(), true, false])(
    "checks if '%p' isn't a IssueSeverity",
    severity => {
      expect(isIssueSeverity(severity)).toEqual(false);
    }
  );

  it.each([
    // ERROR
    [IssueSeverity.ERROR, IssueSeverity.ERROR, 0],
    [IssueSeverity.ERROR, IssueSeverity.WARNING, -1],

    // WARNING
    [IssueSeverity.WARNING, IssueSeverity.ERROR, 1],
    [IssueSeverity.WARNING, IssueSeverity.WARNING, 0]
  ])("compares issue severity '%p' with '%p' and returns '%p'", (...args) => {
    const [severityA, severityB, result] = args as [
      IssueSeverity,
      IssueSeverity,
      number
    ];
    expect(compareIssueSeverities(severityA, severityB)).toEqual(result);
  });
});
