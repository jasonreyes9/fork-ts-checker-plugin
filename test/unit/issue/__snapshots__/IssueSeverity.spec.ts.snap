// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[UNIT] issue/IssueSeverity defines issue severity enum 1`] = `
Object {
  "ERROR": "error",
  "WARNING": "warning",
}
`;
